app_users/views.py

```diff
@@ -103,3 +102,47 @@ def get(self, request, uidb64, token, *args, **kwargs):
         else:
             messages.warning(request, ('The confirmation link was invalid, possibly because it has already been used.'))
             return redirect('homepage')
+
+
+def news_sub(request):
+    form = NewsletterForm()
+    return render(request, 'newsletter.html', {'form': form})
+
+
+def subscribe(request):
+    data = request.GET.get('email')
+    user = User.objects.filter(email__iexact=data)
+    for i in user:
+        if NewsletterSub.objects.get(user=i):
+            NewsletterSub.objects.update(user=i,
+                                         subscribed=True)
+        else:
+            NewsletterSub.objects.create(
+                user=i,
+                subscribed=True
+            )
+    return HttpResponseRedirect('/home')
+
+
+def news_unsub(request):
+    unsub = True
+    return render(request, 'newsletter.html', {'unsubscribe': unsub})
+
+
+def unsubscribe(request):
+    data = request.user
+    user = User.objects.filter(username=data)
+    for i in user:
+        if NewsletterSub.objects.filter(user=i):
+            NewsletterSub.objects.update(
+                user=i,
+                subscribed=False
+            )
```

Comment:
This will update all the objects. I don't think it's what you want. I think you want to filter by user_id first and then update.